Add simplecov

kbrock opened the pull request:
not tested but copied from manageiq

Fryguy commented on Gemfile:
```diff
@@ -12,4 +12,5 @@ gem "manageiq-style"
 gem "rake", "~> 13.0"
 gem "rspec"
 gem "rubocop"
+gem 'simplecov', require: false
```
this should be a dev dependency in the gemspec (in fact all of these should - not sure why they aren't)

Fryguy replied:
Also simplecov needs a minimum for codeclimate - looking at another gem, it's

```
spec.add_development_dependency "simplecov", ">= 0.21.2"
```

Fryguy commented on spec/spec_helper.rb:
```diff
@@ -12,6 +12,14 @@
 # it.
 #
+
+if ENV['CI']
+  require 'simplecov'
+  SimpleCov.start do
+    add_filter "/spec/"
+  end
+end
```
> not tested but copied from manageiq

Probably better to copy from another gem as opposed to an application.

So, here it would just be (I looked at linux_admin as an example)

```suggestion
if ENV['CI']
  require 'simplecov'
  SimpleCov.start
end
```

kbrock replied:
yea. saw no need to run simplecov on specs, since they always display 100%

Fryguy replied:
I was certain I had a script for new repositories and ran it on here.  It should have included all this 😕

Fryguy commented on .github/workflows/ci.yaml:
```diff
@@ -26,3 +26,7 @@ jobs:
+    - name: Report code coverage
+      if: "${{ github.ref == 'refs/heads/master' && matrix.ruby-version == '3.2' }}"
+      continue-on-error: true
+      uses: paambaati/codeclimate-action@v5
```
This won't work without the secret.  Looking at another gem, add this above the steps

```
    env:
      CC_TEST_REPORTER_ID: "${{ secrets.CC_TEST_REPORTER_ID }}"
```

Example: https://github.com/ManageIQ/linux_admin/blob/746d84d93827ef930f31ee7611e93d56c8c83e84/.github/workflows/ci.yaml#L19-L20

kbrock commented:
update:
- moved gemfile dependencies
- fixed simplecov `require`
- added minimum version for simplecov
- added secret

update
- fixed gemspec type
